lib/bolt_spec/run.rb

```diff
@@ -82,8 +82,24 @@ class BoltRunner
       # still be loaded
       def self.with_runner(config_data, inventory_data)
         Dir.mktmpdir do |boltdir_path|
-          config = Bolt::Config.new(Bolt::Boltdir.new(boltdir_path), config_data || {})
-          inventory = Bolt::Inventory.new(inventory_data || {}, config)
+          boltdir = Bolt::Boltdir.new(boltdir_path)
+
+          config = if config_data
+                     Bolt::Config.new(boltdir, config_data)
+                   elsif defined?(RSpec.current_example.example_group_instance.bolt_config)
```

Comment:
I don't think we should rely on RSpec being imported. IMO if any `bolt_config` or `bolt_inventory` is in scope we should just them.

Reply:
I added this to support an RSpec test that looks something like:
```ruby
require 'bolt_spec/run'

describe "my_test" do
  include BoltSpec::Run

  let(:bolt_config) { {'modulepath' => 'foo' } }

  it 'should return' do
    result = run_task('foo', 'bar')
  end
end
```
Are you suggesting implementing a more generic lookup for `bolt_config` to replace the RSpec specific one, or simply adding a new conditional on `defined?(bolt_config)`?

Reply:
something more generic, We want it to work whether rspec is being used or not.